alf/data_structures.py

```diff
@@ -110,6 +111,37 @@ def is_last(self):
         raise ValueError('step_type is not a Torch Tensor')
 
 
+def _create_timestep(observation, prev_action, reward, discount, env_id,
+                     step_type):
+    discount = torch.tensor(discount)
+    # as_tensor reuses the underlying data store of numpy array if possible.
+    create_tensor = lambda t: t.detach() if torch.is_tensor(
+        t) else torch.as_tensor(t).detach()
+    make_tensors = lambda struct: map_structure(create_tensor, struct)
+    return TimeStep(
+        step_type=step_type.view(discount.shape),
```

Comment:
Why do you want to use `view` here? Just `torch.as_tensor(step_type)` is OK

Reply:
This scalar tensor needs to be reshaped, otherwise it cannot be concat with other similar shaped tensors.